lib/faraday/panoptes/client_credentials_authentication.rb

```diff
@@ -45,6 +50,8 @@ def get_token
           grant_type: 'client_credentials',
           client_id: @client_id,
           client_secret: @client_secret)
+
+        raise CredentialsOAuthError.new "Failed to obtain access token" if result.status == 422 or result.body.empty?
```

Comment:
Looks like it can also return `401` if the `client_id` exists but the `client_secret` is wrong. Might want to add that status code here too?